includes/class-sensei-course-structure.php

```diff
@@ -713,7 +720,7 @@ private function validate_item_structure( array $raw_item ) {
 		if ( ! isset( $raw_item['title'] ) || '' === trim( sanitize_text_field( $raw_item['title'] ) ) ) {
 			return new WP_Error(
 				'sensei_course_structure_missing_title',
-				__( 'All items must have a `title` set.', 'sensei-lms' )
+				__( 'Please ensure all items have a name before saving.', 'sensei-lms' )
```

Comment:
The [issue](https://github.com/Automattic/sensei/issues/3693) suggested the message: `Please ensure all modules have a name before saving.`

But this code is testing the lessons and modules to see if one of them doesn't have a name. If we really want to separate this message (for modules and lessons), we can add a new validation to this code, but I didn't add it for now to prevent extra complexity.

Note: The error with the lessons without name won't happen in this case because the frontend already skips lessons without a name, but as it's the API side, it's important to keep it being validated.

Reply:
Yeah that makes sense. It could say `all modules and lessons`, but still it would be clearer on the frontend to only mention modules, to not cause confusion with empty lessons. 
We could handle the error code in the frontend, and display a custom message there for `sensei_course_structure_missing_title` instead of the API-sent message.

Reply:
I separated the modules and lessons error message to be a little more clean and consistent. ;)
https://github.com/Automattic/sensei/pull/3707/commits/f46f5a60a7513d7019a701db2da79db8312a5a9a